terratorch/datasets/sen4agrinet.py

```diff
@@ -0,0 +1,249 @@
+
+        if split == "train":
+            self.image_files = self.train_files
+        elif split == "val":
+            self.image_files = self.val_files
+        elif split == "test":
+            self.image_files = self.test_files
+
+    def __len__(self):
+        return len(self.image_files)
+
+    def split_data(self):
+        random.seed(self.seed)
+
+        if self.scenario == "random":
+            random.shuffle(self.image_files)
+            total_files = len(self.image_files)
+            train_split = int(0.6 * total_files)
+            val_split = int(0.8 * total_files)
+
+            train_files = self.image_files[:train_split]
+            val_files = self.image_files[train_split:val_split]
+            test_files = self.image_files[val_split:]
+
+        elif self.scenario == "spatial":
+            catalonia_files = [f for f in self.image_files if any(tile in f.stem for tile in CAT_TILES)]
+            france_files = [f for f in self.image_files if any(tile in f.stem for tile in FR_TILES)]
+
+            val_split_cat = int(0.2 * len(catalonia_files))
+            train_files = catalonia_files[val_split_cat:]
+            val_files = catalonia_files[:val_split_cat]
+            test_files = france_files
+
+        elif self.scenario == "spatio-temporal":
+            france_files = [f for f in self.image_files if any(tile in f.stem for tile in FR_TILES)]
+            catalonia_files = [f for f in self.image_files if any(tile in f.stem for tile in CAT_TILES)]
+
+            france_2019_files = [f for f in france_files if "2019" in f.stem]
+            catalonia_2020_files = [f for f in catalonia_files if "2020" in f.stem]
+
+            val_split_france_2019 = int(0.2 * len(france_2019_files))
+            train_files = france_2019_files[val_split_france_2019:]
+            val_files = france_2019_files[:val_split_france_2019]
+            test_files = catalonia_2020_files
+
+        return train_files, val_files, test_files
+
+
+    def __getitem__(self, index: int):
+        patch_file = self.image_files[index]
+
+        with h5py.File(patch_file, "r") as patch_data:
+            output = {}
+            images_over_time = []
+            for band in self.bands:
+                band_group = patch_data[band]
+                band_data = band_group[f"{band}"][:]
+
+                band_data = band_data.astype(np.float32)
+
+                if self.truncate_image:
+                    band_data = band_data[-self.truncate_image:]
+                if self.pad_image:
+                    band_data = pad_numpy(band_data, self.pad_image)
+
+                if self.interpolate_and_concat_bands:
+                    band_data = torch.from_numpy(band_data)
+                    band_data = band_data.clone().detach()
+
+                    interpolated = F.interpolate(
+                        band_data.unsqueeze(0), size=(366, 366), mode="bilinear", align_corners=False
```

Comment:
same comment as in the other dataset, is 366 just the size of the largest of all these images? And would it make sense to make this configurable? (maybe it doesnt, curious about your opinion)